Orange/data/sql/backend/mssql.py

```diff
@@ -117,17 +117,26 @@ def _guess_variable(self, field_name, field_metadata, inspect_table):
     EST_ROWS_RE = re.compile(r'StatementEstRows="(\d+)"')
 
     def count_approx(self, query):
-        try:
-            with self.connection.cursor() as cur:
+        with self.connection.cursor() as cur:
+            try:
                 cur.execute("SET SHOWPLAN_XML ON")
                 try:
                     cur.execute(query)
                     result = cur.fetchone()
                     return int(self.EST_ROWS_RE.search(result[0]).group(1))
+                except AttributeError:
+                # This is to catch a float received in StatementEstRows
+                # a float is received when the server's statistics are out of date.
+                    pass
```

Comment:
I have read your comment and I agree, if the float is a result of a reallly bad estimate, it should not be used. I still do not like the flow of this function though. I would put it like this:
```
match = self.EST_ROWS_RE.search(result[0])
if not match:
    # Comment explaining why this happens
    return None
return int(match.group(1))
```